lib/Runtime/Types/PathTypeHandler.h

```diff
@@ -32,19 +32,65 @@ namespace Js
         operator hash_t() const { return GetHashCode(); }
     };
 
-    class SimplePathTypeHandler;
+    class PathTypeSuccessorInfo
+    {
+    public:
+        virtual bool IsSingleSuccessor() const = 0;
```

Comment:
Since there are only two subclass, I wonder whether it is better to just have a "kind" field and use if/else cast instead of virtual, to void a virtual dispatch?

Reply:
Yeah, that should be fine. I don't think it matters much, but a "kind" is probably better.